addons/account/models/account_account_tag.py

```diff
@@ -29,3 +29,9 @@ def _validate_tag_country(self):
         for record in self:
             if record.applicability == 'taxes' and not record.country_id:
                 raise ValidationError(_("A tag defined to be used on taxes must always have a country set."))
+
+    def unlink(self):
```

Comment:
@william-andre I guess you meant "use `@api.ondelete`" ?

Reply:
Well, that would be if we wanted to have custom code for that (instead of `unlink()` indeed).
But the `ondelete` attribute on the field should be able to manage that in a generic way.

Reply:
@william-andre ondelete is not useful here. The error happens on reading record by reference. It's not about some many2one field

To use `@api.ondelete` the PR should be retargeted to v15